openedx/core/djangoapps/import_export/courses/views.py

```diff
@@ -0,0 +1,529 @@
+                redirect_url,
+                urlencode(params)
+            ))
+        else:
+            return JsonResponse(params)
+
+    # pylint: disable=unused-argument
+    @ensure_valid_course_key
+    @renderer_classes_decorator((ArchiveRenderer,))
+    def get(self, request, course_key_string):
+        """
+        The restful handler for exporting a full course or content library.
+
+        GET
+            application/x-tgz: return tar.gz file containing exported course
+            json: not supported
+
+        Note that there are 2 ways to request the tar.gz file. The request
+        header can specify application/x-tgz via HTTP_ACCEPT, or a query
+        parameter can be used (?accept=application/x-tgz).
+
+        If the tar.gz file has been requested but the export operation fails,
+        a JSON string will be returned which describes the error
+        """
+        redirect_url = request.QUERY_PARAMS.get('redirect', None)
+
+        courselike_key = CourseKey.from_string(course_key_string)
+        library = isinstance(courselike_key, LibraryLocator)
+
+        if library:
+            courselike_module = modulestore().get_library(courselike_key)
+        else:
+            courselike_module = modulestore().get_course(courselike_key)
+
+        name = courselike_module.url_name
+        export_file = NamedTemporaryFile(prefix=name + '.', suffix=".tar.gz")
+        root_dir = path(mkdtemp())
+
+        try:
+            if library:
+                export_library_to_xml(
+                    modulestore(),
+                    contentstore(),
+                    courselike_key,
+                    root_dir,
+                    name
+                )
+            else:
+                export_course_to_xml(
+                    modulestore(),
+                    contentstore(),
+                    courselike_module.id,
+                    root_dir,
+                    name
+                )
+
+            logging.debug(
+                u'tar file being generated at {0}'.format(export_file.name)
+            )
+            with tarfile.open(name=export_file.name, mode='w:gz') as tar_file:
+                tar_file.add(root_dir / name, arcname=name)
+        except SerializationError as exc:
+            log.exception(
+                u'There was an error exporting course %s',
+                courselike_key
+            )
+            unit = None
+            failed_item = None
+            parent = None
+            try:
+                failed_item = modulestore().get_item(exc.location)
+                parent_loc = modulestore().get_parent_location(
+                    failed_item.location
+                )
+
+                if parent_loc is not None:
+                    parent = modulestore().get_item(parent_loc)
+                    if parent.location.category == 'vertical':
+                        unit = parent
+            except:  # pylint: disable=bare-except
```

Comment:
I prefer doing
`except Exception:  # pylint: disable=broad-except`

It's a nit, but you're explicitly disabling the more correct thing.